src/main/kotlin/rx/lang/kotlin/observables.kt

```diff
@@ -107,3 +87,12 @@ fun <T, R> List<Observable<T>>.zip(zipFunction: (args: List<T>) -> R): Observabl
  * Returns an Observable that emits the items emitted by the source Observable, converted to the specified type.
  */
 inline fun <reified R : Any> Observable<*>.cast(): Observable<R> = cast(R::class.java)
```

Comment:
I've been thinking about `cast()` and `ofType()` all day. It's a cool demonstration of Kotlin reification, but it kind of feels like a parlor trick with these extension functions. Honestly, it make usage somewhat mysterious and difficult to follow. I believe it is better to be explicit on what you are `ofType()`-ing or `cast()`-ing to. 

Plus, `ofType()` is just the wrong name here. It is not semantically clear in usage. Either way, I think we should remove both.

```kotlin
    @Test
    fun test() {
        val strings: Observable<String> = Observable.just("Alpha", 1, "Beta").ofType()
        strings.subscribe(::println)
    }
```